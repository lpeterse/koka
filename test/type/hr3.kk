val ids = [id : forall<a> a -> a]

fun test1() = head(ids)(1)