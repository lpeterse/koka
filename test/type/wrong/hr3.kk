fun poly(xs : list<forall<a> a -> a>)  {
  1
}

fun wrong(x) {
  poly(x)
}