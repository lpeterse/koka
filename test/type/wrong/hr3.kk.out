test/type/wrong/hr3.kk(5,11): error: unannotated parameters cannot be polymorphic
 context : fun wrong(x) {
 poly(x)
 }
 term : x
 inferred type: list<forall<a> (a) -> a>
 hint : annotate the parameter with a polymorphic type